pyaedt/modeler/circuits/PrimitivesEmit.py

```diff
@@ -771,19 +774,31 @@ def set_band_start_frequency(self, band_node, band_start_freq, units=""):
 
         if not units or units not in emit_consts.EMIT_VALID_UNITS["Frequency"]:
             units = "Hz"
+
         # convert to Hz
-        freq_string = "{}".format(consts.unit_converter(band_start_freq, "Freq", units, "Hz"))
-        prop_list = {"StartFrequency": freq_string}
-        band_node._set_prop_value(prop_list)
+        freq_float_in_Hz = consts.unit_converter(band_start_freq, "Freq", units, "Hz")
+        freq_string = "{}".format(freq_float_in_Hz)
+        if not(1 <= freq_float_in_Hz <= 100000000000):
+            raise ValueError("Frequency should be within 1Hz to 100 GHz.")
+        if float(band_node.props['StopFrequency']) < freq_float_in_Hz:
+            stop_freq = freq_float_in_Hz+1
+            band_node._set_prop_value({'StopFrequency': str(stop_freq)})
+        else:
+            prop_list = {"StartFrequency": freq_string}
+            band_node._set_prop_value(prop_list)
+
 
     def set_band_stop_frequency(self, band_node, band_stop_freq, units=""):
         """Set stop frequency of a given band.
```

Comment:
```suggestion
        """Set stop frequency of the band.
```
You are setting it on a particular instance so "of a given band" isn't needed.